core/dnsserver/onstartup_test.go

```diff
@@ -1,16 +1,22 @@
 package dnsserver
 
-import "testing"
+import (
+	"testing"
+
+	"github.com/coredns/coredns/plugin/pkg/dnsutil"
+)
 
 func TestCheckDomainName(t *testing.T) {
 
 	// Note: compileRegexp() returns regexp-object for RFC1035 preferred syntax.
-	regexpObj := compileRegexp() // regexpObj initialised and declared with a regexp-object for RFC1035 preferred syntax.
+	regexpObj := regex1035PrefSyntax() // regexpObj initialised and declared with a regexp-object for RFC1035 preferred syntax.
 
-	// Below conditions checks for RFC 1035 "preferred syntax"
+	// Below condition checks for root domain-name.
 	if !regexpObj.MatchString(".") {
 		t.Fatalf(`Expected true but received false for "." (root domain)`)
 	}
+
+	// Below conditions checks for RFC 1035 "preferred syntax".
```

Comment:
This comment is pretty much redundant.

Instead, group the tests into two categories, "positive cases" (where you expect to get a match) and "negative cases" (where you don't expect to get a match).